### PR TITLE
Update: Make getRequestDimensions a property on the request

### LINKED ISSUE #1074: Change getRequestDimensions to a request property
Currently `getRequestDimensions` is a function in `chart-utils`, but it should just be a property on the request fragment itself

### PR DESCRIPTION
**Pull request by mattgraham1995**
Resolves #1074 

## Description
`getRequestDimensions` doesn't need to be a separate utility function

## Proposed Changes

- Make a property on the request fragment that will replace `getRequestDimensions`

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/addon/models/bard-request-v2/request.ts [hunk 1]
```diff
@@ -190,6 +190,11 @@ export default class RequestFragment extends Fragment.extend(Validations) implem
     return this.columns.filter(column => column.type === 'timeDimension')[0];
   }
 
+  @computed('timeGrainColumn')
+  get nonTimeGrainDimensions(): ColumnFragment[] {
```
Comment (kevinhinterlong) [inline packages/core/addon/models/bard-request-v2/request.ts:194]: Any other potential names? `groupableDimensions`?

Comment (jkusa) [inline packages/core/addon/models/bard-request-v2/request.ts:194]:
```suggestion
  get nonTimeDimensions(): ColumnFragment[] {
```

File: packages/core/addon/models/bard-request-v2/request.ts [hunk 2]
```diff
@@ -190,6 +190,11 @@ export default class RequestFragment extends Fragment.extend(Validations) implem
     return this.columns.filter(column => column.type === 'timeDimension')[0];
   }
 
+  @computed('timeGrainColumn')
```
Comment (jkusa) [inline packages/core/addon/models/bard-request-v2/request.ts:193]:
```suggestion
  @computed('columns.[]')
```
??

File: packages/core/tests/unit/models/bard-request-v2/request-test.js [hunk 3]
```diff
@@ -433,4 +433,12 @@ module('Unit | Model | Fragment | BardRequest  - Request', function(hooks) {
       'The request model attribute was serialized correctly'
     );
   });
+
+  test('nonTimeDimensions', function(assert) {
+    assert.deepEqual(
+      mockModel.request.nonTimeDimensions,
+      mockModel.request.columns.filter(c => c.type === 'dimension'),
+      'nonTimeDimensions retuns expected dimension columns ignoring the timeDimension and metric columns'
+    );
```
Comment (jkusa) [inline packages/core/tests/unit/models/bard-request-v2/request-test.js:442]: Can we add an assert where we check `nonTimeDimensions` is updated when `columns` is updated?

**Review by kevinhinterlong — APPROVED** [review]

**Review by jkusa — APPROVED** [review]

**Pull request merged**